### PR TITLE
[CI] OSX Conda Build

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context
* Example of building the Mac-OS conda-package through CircleCI.

## How Has This Been Tested
* CircleCI

## Types of changes

- [X] Docs change / refactoring / dependency upgrade
- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -328,6 +328,36 @@ jobs:
...
+            conda init "$(basename "${SHELL}")"
+            source ~/.bash_profile
+            conda activate
+            conda install -y anaconda-client git ninja conda-build pip
+            echo $(which -a python)
+            pip install gitpython
+            cd habitat-sim/conda-build
+            git submodule update --init --recursive --jobs 16
+            export PYTHONIOENCODING="utf-8"
+            python macos_matrix_builder.py --ci_test
```
Comment (Skylion007) [inline .circleci/config.yml:359]: @mosra Any idea why install magnum is causing an ENCODING error on Mac?

File: .circleci/config.yml [hunk 2]
```diff
@@ -328,6 +328,36 @@ jobs:
...
               cd conda-build
               python linux_matrix_builder.py --ci_test
+  test_conda_build_osx:
+    macos:
+      xcode: 11.3.0
+    steps:
+      - checkout:
+          path: ./habitat-sim
+      - run:
+          name: Brew Install
```
Comment (erikwijmans) [inline .circleci/config.yml:338]: No need to install brew here.  conda has cmake, ninja, and ccache and conda build will grab what it needs automatically.  You can just install miniconda directly.

File: .circleci/config.yml [hunk 3]
```diff
@@ -328,6 +328,31 @@ jobs:
...
+      - run:
+          name: Conda Install
+          command: |
+            curl -o ~/miniconda.sh -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+            chmod +x ~/miniconda.sh
+            ~/miniconda.sh -b -p $HOME/miniconda
+            rm ~/miniconda.sh
+            export PATH=$HOME/miniconda/bin:$PATH
+            source ~/.bash_profile
+            conda init "$(basename "${SHELL}")"
```
Comment (erikwijmans) [inline .circleci/config.yml:345]: I think this needs to be before `source ~/.bash_profile`.   Alternatively, you can do `source ~/miniconda/etc/profile.d/conda.sh`

**Comment by Skylion007** [comment]
Well removing the UTF-8 chars worked.

**Review by eundersander — COMMENTED** [review]
Under testing, you mentioned CircleCI. Can you point us to a CircleCI log where you ran this?

File: conda-build/macos_matrix_builder.py [hunk 4]
```diff
@@ -33,22 +33,45 @@ def call(cmd, env=None):
...
+        py_vers = ["3.6", "3.7", "3.8"]
+        bullet_modes = [False, True]
 
     for py_ver, use_bullet in itertools.product(py_vers, bullet_modes):
         env = os.environ.copy()
 
         # including a timestamp in anticipation of nightly builds
         env["VERSION"] = habitat_sim.__version__ + time.strftime(".%Y.%m.%d")
+        if args.nightly:
+            env["VERSION"] = env["VERSION"] + time.strftime(".%Y.%m.%d")
```
Comment (eundersander) [inline conda-build/macos_matrix_builder.py:70]: It seems like we're appending a timestamp twice.

**Review by eundersander — APPROVED** [review]
LGTM!

**Comment by mathfac** [comment]
@Skylion007 do you want to enable it for the version release not only for CI testing purposes like [here](https://github.com/facebookresearch/habitat-sim/blob/master/.circleci/config.yml#L639) but without docker?

**Pull request merged**